nixos/modules/services/printing/cups-pdf.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+let
+
+  inherit (lib.attrsets) filterAttrs getAttr mapAttrs mapAttrsToList;
+  inherit (lib.meta) getExe;
+  inherit (lib.modules) mkAfter mkDefault mkIf;
+  inherit (lib.options) literalExpression mkEnableOption mkOption;
+  inherit (lib.strings) concatStrings;
+  inherit (lib.trivial) pipe;
+  inherit (lib.types) attrsOf int lines nullOr oneOf package path singleLineStr str submodule;
+
+  # cups calls its backends as user `lp` (which is good!),
+  # but cups-pdf wants to be called as `root`,
+  # so it can change ownership of files.
+  # We add a suid wrapper and a wrapper script
+  # to trick cups into calling the suid wrapper.
+
+  # wrapper script that redirects calls to the suid wrapper
+  wrapper = pkgs.writeTextFile {
```

Comment:
```suggestion
  cups-pdf-to-pdf-wrapper = pkgs.writeTextFile {
```

Reply:
I'm just calling this one `cups-pdf-wrapper`, as I attempt to keep this module as general as possible (see also the discussion in https://github.com/NixOS/nixpkgs/pull/182360#discussion_r928051896 ).